tests/src/common/test_case.rs

```diff
@@ -283,17 +286,70 @@ where
     }
 
     pub fn init_profit(&mut self, cadence_hours: u16) -> &mut Self {
-        self.profit_addr = Some(ProfitWrapper::default().instantiate(
-            &mut self.app,
-            cadence_hours,
-            self.treasury_addr.clone().unwrap(),
-            self.timealarms.clone().unwrap(),
-        ));
+        const CONNECTION_ID: &str = "dex-connection";
+
+        let profit_addr: &Addr = self
+            .profit_addr
+            .insert(ProfitWrapper::default().instantiate(
+                &mut self.app,
+                cadence_hours,
+                self.treasury_addr.clone().unwrap(),
+                self.oracle.clone().unwrap(),
+                self.timealarms.clone().unwrap(),
+                CONNECTION_ID.into(),
+            ));
```

Comment:
rebase needed to take this `insert` out?

Reply:
Actually no. That's needed as I later use the address and there's no need to do an `unwrap`.

Reply:
I do not understand why we need to change the `self.profit_addr` in the `init_profit` method. We do want to initialize it.

Reply:
I use it to initialize the contract from start to end. I'm not changing anything about `self.profit_addr` apart from setting it.

Reply:
the code would be more readable to keep the value in a local variable, and then assign a copy to the instance variable

Reply:
it's of your preference so resolving it